### ISSUE TITLE
Cannot reprodue the Bullet Hell example

### ISSUE THREAD

**Issue by chenzhutian** [Unity-Technologies/ml-agents#5701]
**Describe the bug**
Hi, I try to reproduce the dodge bullet example presented in this [release note](https://blog.unity.com/technology/ml-agents-v20-release-now-supports-training-complex-cooperative-behaviors). I get the code from the [dev-bullet-hell](https://github.com/Unity-Technologies/ml-agents/tree/develop-bullet-hell/Project/Assets/ML-Agents/Examples/Bullet) branch. The pretrained model works well  and I can run the training code to start the training. However, I just cannot get the expected reward (the max rewards should be ~5.0, but I only got ~0.2 after training). Can you please help? thanks!

![image](https://user-images.githubusercontent.com/3315274/152806576-8058fcef-58e7-4efe-a974-e4cd4d1b9ca1.png)

I used the training config from [here](https://github.com/Unity-Technologies/ml-agents/blob/bullet-hell-barracuda-test-1.3.1/config/ppo/Bullet.yaml)
```yaml
behaviors:
  Dodge:
    trainer_type: ppo
    hyperparameters:
      batch_size: 1024
      buffer_size: 10240
      learning_rate: 0.0003
      beta: 0.005
      epsilon: 0.2
      lambd: 0.95
      num_epoch: 3
      learning_rate_schedule: linear
    network_settings:
      normalize: true
      hidden_units: 128
      num_layers: 2
      vis_encode_type: simple
    reward_signals:
      extrinsic:
        gamma: 0.99
        strength: 1.0
    keep_checkpoints: 5
    max_steps: 50000000
    time_horizon: 64
    summary_freq: 100000
    threaded: true
```

**Comment by miguelalonsojr** [comment]
Hi! Thanks for reaching out. Unfortunately, this is not an officially supported ML-Agents example environment. As such, it's not actively maintained and is provided as-is. However, it seems that you're using another branch for bullet hell. The one that was used for the blog post is this one: https://github.com/Unity-Technologies/ml-agents/tree/develop-bullet-hell-buffer. Be sure to use that branch for the unity project, the same versions of ml-agents and ml-agents-envs python packages, as well as, the config Bullet.yaml from that branch.

**Comment by chenzhutian** [comment]
Thanks for your reply! 
I found that I posted the wrong link and actually I am using the branch you provide.
However, there is no config file in the branch you provided. Can you please also share the config file? Thanks!

**Comment by chenzhutian** [comment]
I currently use the latest stable version of ml-agents and the code from that branch to train the model. I think it is acceptable if the agent does not perform as good as the one in the blog post. But right now, it seems that the agent learns nothing.

**Comment by miguelalonsojr** [comment]
Hmmm, seems like the config for the bullet hell env was not added to that branch. You can try using this one from a previous bullet hell branch: https://github.com/Unity-Technologies/ml-agents/blob/develop-bullet-hell/config/ppo/Bullet.yaml.

**Comment by miguelalonsojr** [comment]
Also, please use the version of ML-Agents that is on that branch as opposed to the latest stable version. Since this is not an officially supported environment, we have not tested it with the latest stable version of the toolkit. Your best best to reproduce the results from the blog post is to use the exact setup on that branch.

**Comment by chenzhutian** [comment]
Thanks!
I will try to use Release 17 and let you know. Thanks again!

**Comment by chenzhutian** [comment]
Hi, I use exactly the same versions and it turns out the agent can learn something.
However, it take x3 times steps than the config file to make the agent dodge some bullets. And the reward is very unstable.
Do you have any idea about this? Is it because the demo in the blog use another config files with, eg., Curriculum learning or what not?
![Screenshot 2022-02-10 170749](https://user-images.githubusercontent.com/3315274/153491112-7e94d1d8-887a-498a-83f8-f486001afb25.png)
(the rewards)

Thanks!
